### PR TITLE
Repair deferred updates and other cleanup

### LINKED ISSUE #396: Allow feedback.zip to be "not emailed"
Right now when making the feedback.zip we try to email it by default. Instead - provide the user a popup and inform them where the file is, and ask them if they want to try to email the file or not.

Experience should be something like "Feedback zip has been created at: c:\path\to\file." Then let them decide to email it to us - or slack it etc.

### LINKED ISSUE #474: Non-disruptive Ziti Desktop Edge Updates
The auto update / installation process in ziti deskop edge is service disruptive where the ziti service and the ongoing connections are terminated as part of the upgrade process.  The user is  therefore forced to lose their application connections or data rather than being able to choose when they wish to update. Can we do our software update like how windows does it - 1. Allow user to manually update 2. Update at set time / after  - ex after x / y/z/ hours that the user can select 3. Allow it to update as per the notification. As orgs start using us for critical apps / traffic, we should handle updates that can be controlled by users such that it doesn't disrupt their work.

### LINKED ISSUE #541: Automatic update did not provide warning period before updating
see https://openziti.discourse.group/t/updates-for-zedw/599/4  user was USING ziti via ZDEW when 2.0.0 was published and did not get a warning and could not control when the upgrade happened. this is a poor user experience and we need to at least give them time to close out what they are doing. better yet is to let them delay the update by continually snoozing the update etc

### PR DESCRIPTION
**Pull request by dovholuknf**
closes #541
closes #474
closes #396

### CODE REVIEW

File: DesktopEdge/MainWindow.xaml.cs [hunk 1]
```diff
@@ -835,9 +835,18 @@ public partial class MainWindow : Window {
 				if ("installationupdate".Equals(evt.Message?.ToLower())) {
 					logger.Debug("Installation Update is available - {0}", evt.ZDEVersion);
 					IsUpdateAvailable = true;
-					MainMenu.ShowUpdateAvailable(evt.TimeRemaining, evt.InstallTime);
```
Comment (scareything) [inline DesktopEdge/MainWindow.xaml.cs:838]: was TimeRemaining bogus? or does it mean something else? should it be removed?

File: DesktopEdge/Utils/UIUtils.cs [hunk 2]
```diff
@@ -9,8 +9,14 @@ public class UIUtils {
...
 
 		public static void SetLogLevel(string level) {
-			try {
+			try
+			{
 				Logger.Info("request to change log level received: {0}", level);
+				if ((""+level).ToLower() == "verbose")
+                {
+					level = "trace";
+					Logger.Info("request to change log level to verbose - but using trace instead");
```
Comment (scareything) [inline DesktopEdge/Utils/UIUtils.cs:18]: why not respect "verbose" ?

File: DesktopEdge/MainWindow.xaml.cs (hunk 1 above)
Comment (dovholuknf) [reply DesktopEdge/MainWindow.xaml.cs:838]: It was utterly superfluous and i removed it from the json object the montior service sends, yes. It was replaced by a simple "now - installtime" calculation (which is already there):
```
var remaining = evt.InstallTime - DateTime.Now;
...
MainMenu.ShowUpdateAvailable(remaining.TotalSeconds, evt.InstallTime);
```

File: DesktopEdge/Utils/UIUtils.cs (hunk 2 above)
Comment (dovholuknf) [reply DesktopEdge/Utils/UIUtils.cs:18]: NLog doesn't have the level

File: ZitiDesktopEdge.Client/DataStructures/DataStructures.cs [hunk 3]
```diff
@@ -495,10 +494,10 @@ public class StatusCheck : MonitorServiceStatusEvent {
 
     public class InstallationNotificationEvent : MonitorServiceStatusEvent
     {
-        public String ZDEVersion { get; set; }
-        public DateTime CreationDate { get; set; }
+        public string ZDEVersion { get; set; }
+        /*public DateTime CreationDate { get; set; }
```
Comment (dovholuknf) [inline ZitiDesktopEdge.Client/DataStructures/DataStructures.cs:498]: i'll just delete this dead code - this is where TimeRemaining was

File: ZitiUpdateService/App.config [hunk 4]
```diff
@@ -2,17 +2,50 @@
...
+    Format for InstallationReminder is "days:hours:minutes:seconds" - none of the values can exceed the normal range of each tuple.
+    
+    When an installation is detected, the user will be nagged every on this interval until the update is completed. This timer
+    and nagging will be automatic and any clients that attach will get a notification first immediately when attaching to the
+    service. After that they will get a nag reminder based on when the update was published.
+    
+    This duration probably should not be smaller than the UpdateTimer. If a value is provided that is smaller than the UpdateTimer.
+    If the value is smaller than the UpdateTimer duration, it will nag the user every time the update check detects a new udpate.
+    
+    It is recommended to keep this value some multiplier of the UpdateTimer.
```
Comment (scareything) [inline ZitiUpdateService/App.config:30]: `s/multiplier/multiple/` ?

Comment (dovholuknf) [reply ZitiUpdateService/App.config:30]: sure

**Review by scareything — APPROVED** [review]

**Pull request merged**